### PR TITLE
Add pre-commit GitHub workflow, add "Getting started" docs

### LINKED ISSUE #9: Create home page for Eligibility API docs
From https://github.com/cal-itp/eligibility-api/pull/7#discussion_r750512676
> I think we will want the first docs page to be a little more general / about the library, maybe some quick usage samples, etc.

### PR DESCRIPTION
**Pull request by angela-tran**
- Adds GitHub action for pre-commit
- Adds Python linting
- Closes #9

Similiar to cal-itp/eligibility-server#14

### CODE REVIEW

**Comment by angela-tran** [comment]
@machikoyasuda This PR is ready for your review!

**Comment by machikoyasuda** [comment]
@angela-tran Could you add the basic repo set up command instructions? I _think_ I can guess them, but I don't want to mess it up and want to set it up as you've intended!

**Comment by angela-tran** [comment]
@machikoyasuda Sure thing. Just to make sure I understand, do you mean instructions kind of like these: https://docs.calitp.org/eligibility-server/#getting-started-with-the-app?

**Comment by machikoyasuda** [comment]
> @machikoyasuda Sure thing. Just to make sure I understand, do you mean instructions kind of like these: https://docs.calitp.org/eligibility-server/#getting-started-with-the-app?

yes! just the basic commands to clone, build and run the container

**Comment by angela-tran** [comment]
@machikoyasuda I added a home page and a "Getting started" page to the docs, which, if approved, means we can close #9 🎉 

Here is the rendered output that I see locally:
image

You should be able to just open the devcontainer in VS Code without having to build anything beforehand.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "python.linting.flake8Enabled": true,
+    "python.linting.enabled": true,
+    "python.languageServer": "Pylance",
+}
```
Comment (thekaveman) [inline .vscode/settings.json:5]: Here are some additional settings for Python / linting type stuff:

```json
{
    "editor.formatOnSave": true,
    "files.encoding": "utf8",
    "files.eol": "\n",
    "files.insertFinalNewline": true,
    "files.trimFinalNewlines": true,
    "files.trimTrailingWhitespace": true,
    "python.formatting.provider": "black",
}
```

We'll want to add `black` and `flake8` to the [`pip install` in the Dockerfile](https://github.com/cal-itp/eligibility-api/blob/main/.devcontainer/Dockerfile#L18) though

**Review by thekaveman — CHANGES_REQUESTED** [review]

**Review by machikoyasuda — APPROVED** [review]
Once @thekaveman's edits are added, I'm good with this PR!

I ran this locally with the instructions and tested adding a commit that doesn't pass pre-commit rules: 
image
and it worked great.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    "python.linting.enabled": true,
+    "python.languageServer": "Pylance",
+    "editor.formatOnSave": true,
+    "files.encoding": "utf8",
+    "files.eol": "\n",
+    "files.insertFinalNewline": true,
+    "files.trimFinalNewlines": true,
+    "files.trimTrailingWhitespace": true,
+    "python.formatting.provider": "black",
+}
```
Comment (thekaveman) [inline .vscode/settings.json:12]: Nitpick but can we please alphabetize these entries? Makes it easier to scan and find one later

**Review by thekaveman — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 2 above)
Comment (angela-tran) [reply .vscode/settings.json:12]: Sure 👍

**Review by thekaveman — APPROVED** [review]
Nice!

**Pull request merged**